presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java

```diff
@@ -377,21 +378,28 @@ private static SqlTimestamp sqlTimestamp(long millisUtc)
     private void assertCastWithJsonParse(String json, String castSqlType, Type expectedType, Object expected)
     {
         // building queries with VALUES to avoid constant folding
-        MaterializedResult result = runner.execute((new StringBuilder()).append("SELECT CAST(JSON_PARSE(col) AS ").append(castSqlType).append(") FROM (VALUES('").append(json).append("')) AS t(col)").toString());
+        MaterializedResult result = runner.execute(generateCastWithJsonParseQuery(json, castSqlType));
         assertEquals(result.getTypes().size(), 1);
         assertEquals(result.getTypes().get(0), expectedType);
         assertEquals(result.getOnlyValue(), expected);
     }
 
-    private void assertInvalidCastWithJsonParse(String json, String castSqlType, StandardErrorCode errorCode, String errorMessage)
+    private void assertInvalidCastWithJsonParse(String json, String castSqlType, String message)
     {
         try {
-            MaterializedResult a = runner.execute((new StringBuilder()).append("SELECT CAST(JSON_PARSE(col) AS ").append(castSqlType).append(") FROM (VALUES('").append(json).append("')) AS t(col)").toString());
-            fail("expect to fail");
+            runner.execute(generateCastWithJsonParseQuery(json, castSqlType));
+            fail("Expected to throw an INVALID_CAST_ARGUMENT exception");
         }
         catch (PrestoException e) {
-            assertTrue(e.getMessage().startsWith(errorMessage));
-            assertEquals(e.getErrorCode(), errorCode.toErrorCode());
+            assertEquals(e.getErrorCode(), INVALID_CAST_ARGUMENT.toErrorCode());
+            assertEquals(e.getMessage(), message);
         }
     }
+
+    private String generateCastWithJsonParseQuery(String json, String castSqlType)
+    {
+        return "" +
+                "SELECT CAST(JSON_PARSE(col) AS " + castSqlType + ") " +
+                "FROM (VALUES('" + json + "')) AS t(col)";
```

Comment:
I agree with you that `+` is better than `StringBuilder` here. But I would inline `generateCastWithJsonParseQuery`. Having this extracted as a function makes reading the tests harder for a human.